utils/vpr_pbtype_arch_wrapper.py

```diff
@@ -0,0 +1,404 @@
+    return tiles
+
+
+# ABCDEFGHIJKLMNOPQRSTUVWXYZ
+def grid_size(tiles: GridDict) -> Tuple[XPos, YPos]:
+    """Get width and height for a grid dictionary."""
+    width = max(x for x, _ in tiles.keys()) + 1
+    height = max(y for _, y in tiles.keys()) + 1
+    return (width, height)
+
+
+def grid_format(tiles: GridDict) -> str:
+    """Print a nicely formatted string from grid dictionary.
+
+    >>> print(grid_format({
+    ...  (0, 0): "A", (1, 0): "B", (2, 0): "C",
+    ...  (0, 1): "X", (1, 1): "Y", (2, 1): "Z",
+    ... }))
+       012
+     0 ABC
+     1 XYZ
+
+    >>> print(grid_format({
+    ...  (0, 0): "A", (1, 0): "B", (2, 0): "C", (3, 0): "D", (4, 0): "E",
+    ...  (0, 1): "M", (1, 1): "N", (2, 1): "O", (3, 1): "P", (4, 1): "Q",
+    ...  (0, 2): "Z", (1, 2): "Y", (2, 2): "X", (3, 2): "W", (4, 2): "V",
+    ... }))
+       01234
+     0 ABCDE
+     1 MNOPQ
+     2 ZYXWV
+    """
+    width, height = grid_size(tiles)
+
+    s = []
+    # X header
+    s.append("   ")
+    assert width < 10, width
+    for x in range(0, width):
+        s.append(str(x))
+    s.append("\n")
+
+    for y in range(0, height):
+        # Y header
+        s.append("{: 2} ".format(y))
+
+        for x in range(0, width):
+            s.append(str(tiles[(x, y)])[0])
+        s.append('\n')
+    return "".join(s[:-1])
+
+
+def grid_place_in_column(tiles: GridDict, x: XPos, values: List[str]):
+    """Place a list of types centered vertical in a given column.
+
+    Modifies the grid dictionary in place.
+
+    >>> t = grid_new(5, 4)
+    >>> grid_place_in_column(t, 1, ['I'])
+    >>> print(grid_format(t))
+       01234
+     0 .....
+     1 .I...
+     2 .....
+     3 .....
+
+    >>> t = grid_new(5, 4)
+    >>> grid_place_in_column(t, 1, ['I', 'I'])
+    >>> grid_place_in_column(t, 3, ['O'])
+    >>> print(grid_format(t))
+       01234
+     0 .....
+     1 .I.O.
+     2 .I...
+     3 .....
+
+    """
+    width, height = grid_size(tiles)
+
+    start = math.floor((height - len(values)) / 2)
+    for i in range(0, len(values)):
+        tiles[(x, start + i)] = values[i]
+
+
+def grid_generate(input_pins: List[str], output_pins: List[str]) -> GridDict:
+    """Generate a grid dict to fit a set of input_pins and output_pins.
+
+    Generates a 5 width grid with following columns;
+     Column 0 - One input tile per input pins.
+     Column 1 - padding
+     Column 2 - A single tile.
+     Column 3 - padding
+     Column 4 - One output tile per output pins.
+
+    """
+    height = max(len(input_pins), len(output_pins)) + 2
+    width = len(['I', '.', 'T', '.', 'O'])
+    tiles = grid_new(width, height)
+    grid_place_in_column(tiles, 0, ['I'] * len(input_pins))
+    grid_place_in_column(tiles, 2, ['T'] * (height - 2))
+    grid_place_in_column(tiles, 4, ['O'] * len(output_pins))
+    return tiles
+
+
+def layout_xml(arch_xml: ET.Element, pbtype_xml: ET.Element):
+    """Generate a `<layout>` with IBUF and OBUF to match given pb_type."""
```

Comment:
Need to document return type and that arch_xml is mutated.

Reply:
Done.